src/base/utils/os.cpp

```diff
@@ -0,0 +1,419 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "os.h"
+
+#ifdef Q_OS_WIN
+#include <stdexcept>
+
+#include <objbase.h>
+#include <shlobj.h>
+#include <shobjidl.h>
+
+#include <boost/intrusive_ptr.hpp>
+
+#include <QDir>
+#include <QSettings>
+#include <QSysInfo>
+
+#include <base/global.h>
+
+#define __WIDE(str) L ## str
+#define WIDE(str) __WIDE(str)
+
+#define APP_REGISTRY_NAME WIDE(QBT_APP_NAME)
+
+namespace
+{
+    const QString CLASS_TORRENT {QLatin1String(C_TORRENT_FILE_EXTENSION)};
+    const QString CLASS_MAGNET {QStringLiteral("magnet")};
+    const QString PROGID_TORRENT {QStringLiteral("qBittorrent.File.Torrent")};
+    const QString PROGID_MAGNET {QStringLiteral("qBittorrent.Url.Magnet")};
+
+    void intrusive_ptr_add_ref(IUnknown *pCOMObj)
+    {
+        pCOMObj->AddRef();
+    }
+
+    void intrusive_ptr_release(IUnknown *pCOMObj)
+    {
+        pCOMObj->Release();
+    }
+
+    class COMLib final
+    {
+        Q_DISABLE_COPY(COMLib)
+
+    public:
+        COMLib(DWORD dwCoInit)
+        {
+            if (FAILED(::CoInitializeEx(NULL, dwCoInit)))
+                throw std::runtime_error("Couldn't initialize COM library.");
+        }
+
+        ~COMLib()
+        {
+            ::CoUninitialize();
+        }
+
+        template <typename T>
+        boost::intrusive_ptr<T> createObject(
+                REFCLSID rclsid, LPUNKNOWN pUnkOuter
+                , DWORD dwClsContext, HRESULT *pError = nullptr) const
+        {
+            T *obj = nullptr;
+            HRESULT result = ::CoCreateInstance(rclsid, pUnkOuter, dwClsContext, __uuidof(T)
+                                                , reinterpret_cast<void**>(&obj));
+            if (pError)
+                *pError = result;
+
+            return {obj, false};
+        }
+    };
+
+    enum REG_SEARCH_TYPE
+    {
+        USER,
+        SYSTEM_32BIT,
+        SYSTEM_64BIT
+    };
+
+    QStringList getRegSubkeys(HKEY handle)
```

Comment:
I would add const to `handle`.

Reply:
I wouldn't want to change this existing code in this PR. I just moved it here as is (I mean all python and windows registry related code).

Reply:
Also I would stop any conversations here until PR will be generally approved by @sledgehammer999. I feel I make unnecessary work...

Reply:
ok, although I wish to see this merged.

Reply:
If this lands, I'm going to submit a PR adding CPack support to build Windows installers (probably via Wix) and Mac OS bundles.